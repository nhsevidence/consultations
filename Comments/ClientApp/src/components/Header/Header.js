// @flow

import React, { Fragment } from "react";
import Moment from "react-moment";

type PropsType = {
	title: string,
	reference: string,
	endDate: any,
	match: any,
	onNewCommentClick: Function
}

export const Header = (props: PropsType) => {
	const title = props.title;
	const reference = props.reference;
	const endDate = props.endDate;

	return (
		<Fragment>
			<h1 className="page-header__heading mt--0">{title}</h1>
			<p className="page-header__lead">
				[{reference}] Open until{" "}
				<Moment format="D MMMM YYYY" date={endDate} />
			</p>
		</ Fragment>
	);
};