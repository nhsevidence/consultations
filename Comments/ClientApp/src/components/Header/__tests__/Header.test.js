import React from "react";
import { shallow } from "enzyme";

import { Header } from "../Header";
import toJson from "enzyme-to-json";

describe("[ClientApp] ", () => {
	describe("Header ", () => {
		it("should display correctly when consulation hasn't started", () => {
			const FakeProps = {
				title: "My title",
				subtitle1: "Subtitle 1",
				subtitle2: "Subtitle 2",
				consultationState: {
					endDate: "1994-11-05T08:15:30-05:00",
					consultationIsOpen: false,
					consultationHasNotStartedYet: true,
				},
			};
			const wrapper = shallow(
				<Header {...FakeProps}/>
			);
			expect(
				toJson(wrapper, {
					noKey: true,
					mode: "deep",
				})
			).toMatchSnapshot();
		});

		it("should display correctly when consulation is in progress", () => {
			const FakeProps = {
				title: "My title",
				subtitle1: "Subtitle 1",
				consultationState: {
					endDate: "1994-11-05T08:15:30-05:00",
					consultationIsOpen: true,
					consultationHasNotStartedYet: false,
				},
			};
			const wrapper = shallow(
				<Header {...FakeProps}/>
			);
			expect(
				toJson(wrapper, {
					noKey: true,
					mode: "deep",
				})
			).toMatchSnapshot();
		});

		it("should display correctly when consulation has ended", () => {
			const FakeProps = {
				title: "My title has changed",
				subtitle1: "Subtitle 1",
				consultationState: {
					endDate: "1994-11-05T08:15:30-05:00",
					consultationIsOpen: false,
					consultationHasNotStartedYet: false,
				},
			};
			const wrapper = shallow(
				<Header {...FakeProps}/>
			);
			expect(
				toJson(wrapper, {
					noKey: true,
					mode: "deep",
				})
			).toMatchSnapshot();
		});
	});
});
