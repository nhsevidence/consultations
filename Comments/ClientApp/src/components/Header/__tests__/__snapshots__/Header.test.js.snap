// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ClientApp]  Header  should display correctly when consulation has ended 1`] = `
<Fragment>
  <h1
    className="page-header__heading mt--0"
    data-qa-sel="changeable-page-header"
  >
    My title has changed
  </h1>
  <p
    className="page-header__lead mb--d"
  >
    This consultation 
    ended
     on 
    <t
      ago={false}
      calendar={false}
      date="1994-11-05T08:15:30-05:00"
      decimal={false}
      element="time"
      filter={[Function]}
      format="D MMMM YYYY"
      fromNow={false}
      interval={60000}
      onChange={[Function]}
      toNow={false}
      unit={null}
      unix={false}
      utc={false}
    />
     at 
     
    <t
      ago={false}
      calendar={false}
      date="1994-11-05T08:15:30-05:00"
      decimal={false}
      element="time"
      filter={[Function]}
      format="HH:mm"
      fromNow={false}
      interval={60000}
      onChange={[Function]}
      toNow={false}
      unit={null}
      unix={false}
      utc={false}
    />
  </p>
  <p>
    Subtitle 1
  </p>
</Fragment>
`;

exports[`[ClientApp]  Header  should display correctly when consulation hasn't started 1`] = `
<Fragment>
  <h1
    className="page-header__heading mt--0"
    data-qa-sel="changeable-page-header"
  >
    My title
  </h1>
  <p
    className="page-header__lead mb--d"
  >
    This consultation 
    starts
     on 
    <t
      ago={false}
      calendar={false}
      date="1994-11-05T08:15:30-05:00"
      decimal={false}
      element="time"
      filter={[Function]}
      format="D MMMM YYYY"
      fromNow={false}
      interval={60000}
      onChange={[Function]}
      toNow={false}
      unit={null}
      unix={false}
      utc={false}
    />
     at 
     
    <t
      ago={false}
      calendar={false}
      date="1994-11-05T08:15:30-05:00"
      decimal={false}
      element="time"
      filter={[Function]}
      format="HH:mm"
      fromNow={false}
      interval={60000}
      onChange={[Function]}
      toNow={false}
      unit={null}
      unix={false}
      utc={false}
    />
  </p>
  <p>
    Subtitle 1
  </p>
  <p>
    Subtitle 2
  </p>
</Fragment>
`;

exports[`[ClientApp]  Header  should display correctly when consulation is in progress 1`] = `
<Fragment>
  <h1
    className="page-header__heading mt--0"
    data-qa-sel="changeable-page-header"
  >
    My title
  </h1>
  <p
    className="page-header__lead mb--d"
  >
    Open until
     
    <t
      ago={false}
      calendar={false}
      date="1994-11-05T08:15:30-05:00"
      decimal={false}
      element="time"
      filter={[Function]}
      format="D MMMM YYYY"
      fromNow={false}
      interval={60000}
      onChange={[Function]}
      toNow={false}
      unit={null}
      unix={false}
      utc={false}
    />
  </p>
  <p>
    Subtitle 1
  </p>
</Fragment>
`;
